idaes/generic_models/properties/core/pure/RPP5.py

```diff
@@ -0,0 +1,190 @@
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Methods for calculating pure component properties from:
+
+The Properties of Gases & Liquids, 5th Edition
+Reid, Prausnitz and Polling, 2001, McGraw-Hill
+
+All parameter indicies based on conventions used by the source
+"""
+
+from pyomo.environ import exp, value, log, Var, units as pyunits
+
+from idaes.core.util.misc import set_param_from_config
+
+from idaes.core.util.constants import Constants as const
+
+# -----------------------------------------------------------------------------
+# Heat capacities, enthalpies and entropies
+class cp_mol_ig_comp():
+
+    @staticmethod
+    def build_parameters(cobj):
+        cobj.cp_mol_ig_comp_coeff_a0 = Var(
+                doc="Parameter a0 for ideal gas molar heat capacity",
+                units=None)
+        set_param_from_config(cobj, param="cp_mol_ig_comp_coeff", index="a0")
+
+        cobj.cp_mol_ig_comp_coeff_a1 = Var(
+                doc="Parameter a1 for ideal gas molar heat capacity",
+                units=pyunits.K**-1)
+        set_param_from_config(cobj, param="cp_mol_ig_comp_coeff", index="a1")
+
+        cobj.cp_mol_ig_comp_coeff_a2 = Var(
+                doc="Parameter a2 for ideal gas molar heat capacity",
+                units=pyunits.K**-2)
+        set_param_from_config(cobj, param="cp_mol_ig_comp_coeff", index="a2")
+
+        cobj.cp_mol_ig_comp_coeff_a3 = Var(
+                doc="Parameter a3 for ideal gas molar heat capacity",
+                units=pyunits.K**-3)
+        set_param_from_config(cobj, param="cp_mol_ig_comp_coeff", index="a3")
+
+        cobj.cp_mol_ig_comp_coeff_a4 = Var(
+                doc="Parameter a4 for ideal gas molar heat capacity",
+                units=pyunits.K**-4)
+        set_param_from_config(cobj, param="cp_mol_ig_comp_coeff", index="a4")
+
+    @staticmethod
+    def return_expression(b, cobj, T):
+        # Specific heat capacity
+        T = pyunits.convert(T, to_units=pyunits.K)
+
+        cp = ((cobj.cp_mol_ig_comp_coeff_a4*T**4 +
+              cobj.cp_mol_ig_comp_coeff_a3*T**3 +
+              cobj.cp_mol_ig_comp_coeff_a2*T**2 +
+              cobj.cp_mol_ig_comp_coeff_a1*T +
+              cobj.cp_mol_ig_comp_coeff_a0) * const.gas_constant)
+
+        units = b.params.get_metadata().derived_units
+        return pyunits.convert(cp, units["heat_capacity_mole"])
+
+
+class enth_mol_ig_comp():
+
+    @staticmethod
+    def build_parameters(cobj):
+        if not hasattr(cobj, "cp_mol_ig_comp_coeff_A"):
+            cp_mol_ig_comp.build_parameters(cobj)
+
+        units = cobj.parent_block().get_metadata().derived_units
+
+        cobj.enth_mol_form_vap_comp_ref = Var(
```

Comment:
Note that I have a PR (#141) waiting to be merged that adds a configuration argument for whether we build and include the heat of formation. We will probably run into some conflicts when we try merging this, and we will want to update this to match the other files as well.